tests/semantic_analyser.cpp

```diff
@@ -3121,23 +3121,49 @@ TEST_F(SemanticAnalyserTest, cast_sign)
   EXPECT_EQ(CreateUInt64(), ul->var()->var_type);
 }
 
-TEST_F(SemanticAnalyserTest, binop_sign)
+TEST_F(SemanticAnalyserTest, binop_bool_and_int)
+{
+  std::string operators[] = {
+    "==", "!=", "<", "<=", ">",  ">=", "&&", "||", "+",
+    "-",  "/",  "*", ">>", "<<", "&",  "|",  "^",
+  };
+
+  // Making a single variable below so as not to trigger fold_literals code
```

Comment:
Can you add tests to the fold literals suite to ensure that bool is covered? (Specifically it would be good to make sure the overflow/underflow cases are handled.)

Reply:
Done.